src/LiveDevelopment/main.js

```diff
@@ -316,6 +316,70 @@ define(function main(require, exports, module) {
         return false;
     }
 
+    // default mode means on first load for pro user we have edit mode
+    // for free user we have highlight mode
+    function _getDefaultMode() {
+        return isProUser ? "edit" : "highlight";
+    }
+
+    // to set that mode in the preferences
+    function _initializeMode() {
+        if (isFreeTrialUser) {
+            PreferencesManager.set(PREFERENCE_LIVE_PREVIEW_MODE, "edit");
+            return;
+        }
+
+        const savedMode = PreferencesManager.get(PREFERENCE_LIVE_PREVIEW_MODE) || _getDefaultMode();
+
+        if (savedMode === "highlight" && isProUser) {
+            PreferencesManager.set(PREFERENCE_LIVE_PREVIEW_MODE, "edit");
+        } else if (savedMode === "edit" && !isProUser) {
+            PreferencesManager.set(PREFERENCE_LIVE_PREVIEW_MODE, "highlight");
+        }
+    }
+
+    // this is called everytime there is a change in entitlements
+    async function _updateProUserStatus() {
+        if (!KernalModeTrust) {
+            return;
+        }
+
+        try {
+            const entitlement = await KernalModeTrust.EntitlementsManager.getLiveEditEntitlement();
+            const isPaidSub = await KernalModeTrust.EntitlementsManager.isPaidSubscriber();
```

Comment:
This check is not needed. Fully depend on activated state of live edit entitlement. If there are other paid plans in future without. Live edit tjis may cause issues.